src/NServiceBus.NHibernate/TimeoutPersisters/TimeoutPersister.cs

```diff
@@ -86,23 +86,23 @@ public Task<TimeoutsChunk> GetNextChunk(DateTime startSlice)
 
         public async Task Add(TimeoutData timeout, ContextBag context)
         {
-            var timeoutId = Guid.NewGuid();
             using (var session = await OpenSession(context).ConfigureAwait(false))
             {
-                session.Session().Save(new TimeoutEntity
+                var timeoutEntity = new TimeoutEntity
                 {
-                    Id = timeoutId,
                     Destination = timeout.Destination,
                     SagaId = timeout.SagaId,
                     State = timeout.State,
                     Time = timeout.Time,
                     Headers = ConvertDictionaryToString(timeout.Headers),
                     Endpoint = timeout.OwningTimeoutManager
-                });
+                };
+
+                session.Session().Save(timeoutEntity);
```

Comment:
Save returns the ID. I think it would be move obvious to use this one instead of getting it again from the property on the entity.